Source/WTF/wtf/LLVMProfileUtils.h

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2025  All rights reserved.
+
+#pragma once
+
+#if ENABLE(LLVM_PROFILE_GENERATION) && PLATFORM(IOS_FAMILY)
+#include "text/MakeString.h"
+#include "text/WTFString.h"
+
+#include <notify.h>
+#include <unistd.h>
+
+extern "C" int __llvm_profile_runtime = 0;
+extern "C" void __llvm_profile_initialize_file(void);
+extern "C" const char *__llvm_profile_get_filename(void);
+
+namespace WTF {
+
+String profileCollectionNotificationNameForCurrentProcess()
+{
+    static auto notificationName = makeString("com.apple.WebKit.llvmProfileInitialize."_s, String::number(getpid()));
+    return notificationName;
+}
+
+template<typename OnceFlagDiversifier>
+ALWAYS_INLINE void setupProfileCollection(const char *name)
+{
+    static std::once_flag registerFlag;
+    std::call_once(registerFlag, [name] {
+        int token;
+        auto notificationName = profileCollectionNotificationNameForCurrentProcess().utf8().data();
+
+        int status = notify_register_dispatch(notificationName, &token, dispatch_get_main_queue(), ^(int) {
+            __llvm_profile_initialize_file();
+            const char *filename = __llvm_profile_get_filename();
+            if (!access(filename, F_OK))
```

Comment:
Nit - not sure the local variable adds much here.

```suggestion
            if (!access(__llvm_profile_get_filename(), F_OK))
```